### PR TITLE
6 tasks section

### PR DESCRIPTION
**Pull request by ewaczajka**
## Changes
- added the Tasks Section with a full logic
- after added a new task you can edit deadline, add assigned users (or click on user logo to remove) and add some extra text 
- click outside component saves changes and closes editor
- click on any checkbox is independent of the opened editor and directly saves that information in a firestore document.

## Not working
- saving the modification data (time, user)
- the check icon is not visible

## To do
- styles refactor, add media queries
- add useCollectionQueries to Notes and FamilyMembers

### CODE REVIEW

File: src/components/molecules/Tasks/Tasks.js [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+    const refCheckboxes = useRef([])
+
+    const [openDetails, setOpenDetails] = useState(false)
+    const [initialValues, setInitialValues] = useState({})
+
+    const [editedTask, dispatch] = useReducer(reducer, {})
+    const [checkedTask, setCheckedTask] = useState({})
+
+    useEffect(() => {
+        const handleClickOutside = e => {
```
Comment (nataliawalasik) [inline src/components/molecules/Tasks/Tasks.js:62]: Someone smart wrote [this](https://usehooks.com/useOnClickOutside/) hook. 
I personally always use it, along with my coworkers

File: src/components/molecules/Tasks/Tasks.js [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+import { TasksOrderRules } from 'data/orderRules'
+
+const taskActionTypes = {
+    addInitialValues: 'ADD INITIAL VALUES',
+    taskChange: 'HANDLE TASK CHANGE',
+    checkedToggle: 'CHECKED TOGGLE',
+    clearValues: 'CLEAR VALUES',
+}
+
+const reducer = (state, action) => {
```
Comment (nataliawalasik) [inline src/components/molecules/Tasks/Tasks.js:23]: The reducer looks so nice ;)

File: src/components/molecules/TasksAddField/TasksAddField.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    const updateCreationData = () => {
+        setNewTask(prevState => ({
+            ...prevState,
+            creationDate: Timestamp.now(),
+            modificationDate: Timestamp.now(),
+        }))
+    }
+
+    return (
+        <AddTaskBox as="form" onSubmit={e => addNewTask(e)}>
```
Comment (nataliawalasik) [inline src/components/molecules/TasksAddField/TasksAddField.js:65]: onSubmit={addNewTask}

File: src/components/molecules/Modal/Modal.styles.js [hunk 4]
```diff
@@ -10,4 +10,5 @@ export const ModalWrapper = styled.div`
     background: ${({ theme }) => theme.background.beige};
     box-shadow: ${({ theme }) => theme.boxShadow.shadow};
     border-radius: 5px;
+    z-index: 9000;
```
[the remark this answers is not in the document]
Comment (ewaczajka) [reply src/components/molecules/Modal/Modal.styles.js:13]: It's already displaying in Portal

File: src/components/molecules/Tasks/Tasks.js (hunk 1 above)
Comment (ewaczajka) [reply src/components/molecules/Tasks/Tasks.js]: I tried to adjust it to my case - is it ok?

File: src/components/molecules/TasksAddField/TasksAddField.js (hunk 3 above)
Comment (ewaczajka) [reply src/components/molecules/TasksAddField/TasksAddField.js]: done

File: src/components/molecules/Tasks/Tasks.js (hunk 1 above)
Comment (ewaczajka) [reply src/components/molecules/Tasks/Tasks.js]: Done!

**Pull request merged**